File: gracie.lee/src/main/java/homework/chapter_6/homework3/Order.java
package homework.chapter_6.homework3;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity(name = "Hw3Order")
@Table(name = "orders_hw3")
public class Order {

    @Id
    @GeneratedValue
    @Column(name = "order_id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "member_id")
    private Member member;  // 다대다를 중간 엔티티로 풀기: Member → Order

    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;  // 다대다를 중간 엔티티로 풀기: Product → Order

    private int count;              // 주문 수량

    private LocalDateTime orderDate; // 주문 일자

    protected Order() {
    }

    public Long getId() {
        return id;
    }

    public Member getMember() {
        return member;
    }

    public Product getProduct() {
        return product;
    }

    public int getCount() {
        return count;
    }

    public LocalDateTime getOrderDate() {
        return orderDate;
    }

    public void setMember(Member member) {
        this.member = member;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public void setOrderDate(LocalDateTime orderDate) {
        this.orderDate = orderDate;
    }
}

File: hyun.k/hyun.k/src/main/kotlin/chapter6/homework3/Order.kt
package chapter6.homework3

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity(name = "Hw3OrderKt")
@Table(name = "orders_hw3_kt")
data class Order(
    @Id
    @GeneratedValue
    @Column(name = "order_id")
    val id: Long? = null,

    @ManyToOne
    @JoinColumn(name = "member_id")
    var member: Member3? = null,  // 다대다를 중간 엔티티로 풀기: Member -> Order

    @ManyToOne
    @JoinColumn(name = "product_id")
    var product: Product? = null,  // 다대다를 중간 엔티티로 풀기: Product -> Order

    var count: Int = 0,              // 주문 수량

    var orderDate: LocalDateTime? = null  // 주문 일자
)